src/components/EnemyCollision.js

```diff
@@ -0,0 +1,40 @@
+import AFRAME from 'aframe';
+
+AFRAME.registerComponent('enemy-collision', {
+  play: function () {
+    let hit = false
+    // check for collision
+    this.el.addEventListener('collide', function (e) {
+        setTimeout(function() {
```

Comment:
why are you using a set timeout here? Is there some race condition?

Reply:
Ya if the timeout isn't there the program doesn't recognize the thrown tiles as part of the projectile pool so with some tiles you get a warning and the tiles don't disappear on impact. I think it has to do with giving the program time to fully load in the components within the collision code.